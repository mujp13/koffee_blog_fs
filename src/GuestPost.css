.guest-post {
  padding: 20px;
}
