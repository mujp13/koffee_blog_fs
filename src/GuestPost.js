import React from 'react';
import './GuestPost.css';

export default function GuestPost(props) {
  return (
    <div className="guest-post">
      <h3>{props.name}</h3>
      <p>{props.comment}</p>
      <p>{props.date.toString()}</p>
    </div>
  );
}
