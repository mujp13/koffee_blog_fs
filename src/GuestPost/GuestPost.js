import React, { Component } from 'react';
import './GuestPost.css';

export default class GuestPost extends Component {
  render() {
    //const { guestbooks } = this.context;
    return (
      <div className="guest-post">
        <h3>{this.props.name}</h3>
        <p>{this.props.comment}</p>
        <p>{this.props.date}</p>
      </div>
    );
  }
}
