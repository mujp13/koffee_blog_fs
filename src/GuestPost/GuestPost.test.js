import React from 'react';
import ReactDOM from 'react-dom';
import GuestPost from './GuestPost';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <GuestPost />
    </MemoryRouter>,
    div
  );

  ReactDOM.unmountComponentAtNode(div);
});

//snapshot
it('renders correctly', () => {
  const tree = renderer.create(<GuestPost />).toJSON();
  expect(tree).toMatchSnapshot();
});
