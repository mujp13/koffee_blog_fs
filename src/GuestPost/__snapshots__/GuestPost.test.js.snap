// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="guest-post"
>
  <h3 />
  <p />
  <p />
</div>
`;
